### PR TITLE
Add status command to the boundary client cache

### PR DESCRIPTION
**Pull request by talanknight**
Sample output:

```
> boundary daemon status

Status:
  Domain Socket:       /Users/myname/.boundary/socket/daemon.sock
  Uptime:              8m9s
  User Count:          2

  User:
    Address:               http://127.0.0.1:9200
    AuthToken Count:       1
    Id:                    u_1234567890
    AuthToken:
      Id:                      at_YdVyvcFO0h
      In Keyring:              true
    Target:
      Count:                    2
      Since Full Refresh:       8m9s
      Since Last Refresh:       10s
    Session:
      Count:                   0
  User:
    Address:               http://127.0.0.1:9200
    AuthToken Count:       1
    Id:                    u_0987654321
    AuthToken:
      Id:                      at_l6vMxulkKj
    Target:
      Count:                    2
      Since Full Refresh:       55s
      Since Last Refresh:       10s
    Session:
      Count:                   0
```

### CODE REVIEW

**Review by jimlambrt — APPROVED** [review]
ty!

File: internal/clientcache/internal/daemon/status_handler.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+type ErrorStatus struct {
+	Error        string
+	LastReturned time.Duration
+}
+
+// ResourceStatus contains the status of a specific resource type contained in
+// the cache for a specific user.
+type ResourceStatus struct {
+	Name         string
```
Comment (jefferai) [inline internal/clientcache/internal/daemon/status_handler.go:31]: Please add `json` tags to the various structs here and anywhere else we are marshaling as the default JSON output does not match our standard casing.

File: internal/clientcache/internal/daemon/status_handler.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	// The boundary address for this user
+	Address string
+	// The auth tokens used by this user to authenticate with the boundary instance
+	AuthTokens []AuthTokenStatus
+	// The resources tracked by the cache for this user
+	Resources []ResourceStatus
+}
+
+// StatusResult is the struct returned to status requests.
+type StatusResult struct {
```
Comment (jefferai) [inline internal/clientcache/internal/daemon/status_handler.go:58]: Given that there will never be anything marshaled as JSON do you even need the tags here at all?

Comment (jefferai) [reply internal/clientcache/internal/daemon/status_handler.go:58]: Oops, sorry, `,omitempty` not `-,omitempty`. But please fix the names :-)

File: internal/clientcache/cmd/daemon/status.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+	case "json":
+		c.UI.Output(resp.Body.String())
+	default:
+		c.UI.Output(printStatusTable(res))
+	}
+	return base.CommandSuccess
+}
+
+func (c *StatusCommand) Status(ctx context.Context) (*api.Response, error) {
+	const op = "daemon.(StatusCommand).Status"
```
Comment (jefferai) [inline internal/clientcache/cmd/daemon/status.go:100]: This const is unused

File: internal/clientcache/cmd/daemon/status.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+	}
+
+	return status(ctx, dotPath)
+}
+
+func status(ctx context.Context, daemonPath string) (*api.Response, error) {
+	const op = "daemon.status"
+	client, err := api.NewClient(nil)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
```
Comment (jefferai) [inline internal/clientcache/cmd/daemon/status.go:113]: These errors will bubble up to printing directly to CLI. We don't return raw errors for CLI commands; they are human readable, capitalized with punctuation (but can also include raw error strings). For example: `c.PrintCliError(fmt.Errorf("Error creating API client: %w", err))`

Comment (jefferai) [reply internal/clientcache/cmd/daemon/status.go:113]: Also we don't include the op anywhere in the CLI at this point. Maybe we should but I don't think we should introduce that pattern here.

File: internal/clientcache/cmd/daemon/status.go [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+	addr := daemon.SocketAddress(daemonPath)
+	_, err = os.Stat(strings.TrimPrefix(addr, "unix://"))
+	if strings.HasPrefix(addr, "unix://") && err != nil {
+		return nil, errDaemonNotRunning
+	}
+	if err := client.SetAddr(addr); err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	// Because this is using the real lib it can pick up from stored locations
+	// like the system keychain. Explicitly clear the token for now
```
Comment (jefferai) [inline internal/clientcache/cmd/daemon/status.go:124]: I'm a bit confused by this. Why are we using the Boundary API client? Does it actually make sense for making simple calls across a unix socket?

File: internal/clientcache/internal/daemon/status_handler.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+	KeyringlessReferences int    `json:"keyringless_references,omitempty"`
+}
+
+// UserStatus contains the status of a specific user tracked by the cache
+type UserStatus struct {
+	// The Id of the user this status is for
+	Id string `json:"id"`
+	// The boundary address for this user
+	Address string `json:"address"`
+	// The auth tokens used by this user to authenticate with the boundary instance
```
Comment (jefferai) [inline internal/clientcache/internal/daemon/status_handler.go:51]: Curious in this file generally why some things are omitempty and some aren't. Is it that the ones not marked omitempty are never expected to be optional?

**Review by jefferai — COMMENTED** [review]
Looking pretty good, just a few more comments

File: internal/clientcache/internal/daemon/status_handler.go [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+type ErrorStatus struct {
+	Error        string        `json:"error,omitempty"`
+	LastReturned time.Duration `json:"last_returned,omitempty"`
+}
+
+// ResourceStatus contains the status of a specific resource type contained in
+// the cache for a specific user.
+type ResourceStatus struct {
+	Name         string              `json:"name,omitempty"`
+	Count        int                 `json:"count"`
```
Comment (jefferai) [inline internal/clientcache/internal/daemon/status_handler.go:32]: Missing `omitempty`?

**Review by jefferai — APPROVED** [review]
Left one more comment!

**Pull request merged**